src/index/TextIndexReadWrite.h

```diff
@@ -341,15 +343,25 @@ requires std::is_arithmetic_v<T> class GapEncode {
   GapEncode(GapEncode&&) = delete;
   GapEncode& operator=(GapEncode&&) = delete;
 
-  void writeToFile(ad_utility::File& out, off_t& currentOffset);
+  void writeToFile(ad_utility::File& out, off_t& currentOffset) {
+    textIndexReadWrite::encodeAndWriteSpanAndMoveOffset<T>(encodedVector_, out,
+                                                           currentOffset);
+  }
 
   const TypedVector& getEncodedVector() const { return encodedVector_; }
 
  private:
   // This method implements the constructor. The reason is explained in the
   // comment above the constructor.
   template <typename View>
-  void initialize(View&& view);
+  void initialize(View&& view) {
+    if (ql::ranges::empty(view)) {
+      return;
+    }
+    encodedVector_.reserve(ql::ranges::size(view));
+    std::adjacent_difference(ql::ranges::begin(view), ql::ranges::end(view),
+                             std::back_inserter(encodedVector_));
+  }
```

Comment:
With the change from above you can put them back to where they were in the .cpp file